src/App.test.tsx

```diff
@@ -1,19 +1,21 @@
-import { render, screen } from '@testing-library/react';
-import userEvent from '@testing-library/user-event';
-import App from './App';
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import App from "./App";
 
-describe('main page', () => {
-  it('renders page heading', async () => {
+describe("main page", () => {
+  it("renders page heading", async () => {
     render(<App />);
-    const heading = await screen.findByRole('heading', { name: 'City List' });
+    const heading = await screen.findByRole("heading", { name: "City List" });
     expect(heading).toBeInTheDocument();
   });
 
-  it('does a search correctly', async () => {
+  it("does a search correctly", async () => {
     render(<App />);
     expect(await screen.findByText(/Tokyo/)).toBeInTheDocument();
-    const textInput = screen.getByRole('textbox', { name: 'Search' });
-    userEvent.type(textInput, 'osaka');
-    expect(screen.queryByText(/Tokyo/)).not.toBeInTheDocument();
+    const textInput = screen.getByRole("textbox");
+    userEvent.type(textInput, "osaka");
+    await waitFor(() => {
+      expect(screen.queryByText(/Tokyo/)).not.toBeInTheDocument();
+    });
```

Comment:
This should be like this because every `userEvent` method is async.
```suggestion
    await userEvent.type(textInput, "osaka");
    expect(screen.queryByText(/Tokyo/)).not.toBeInTheDocument();
```